include/onnxruntime/core/providers/nnapi/nnapi_provider_factory.h

```diff
@@ -1,15 +1,34 @@
 // Copyright 2019 JD.com Inc. JD AI
+#pragma once
 
 #include "onnxruntime_c_api.h"
 
+// NNAPIFlag are bool options we want to set for NNAPI EP
+// To generate a unsigned long nnapi_flags for using with OrtSessionOptionsAppendExecutionProvider_Nnapi below,
+// If using C
+//   unsigned long nnapi_flags = 0;
+//   nnapi_flags |= 1UL << NNAPI_FLAG_USE_FP16;
+// If using C++
+//   std::bitset<NNAPI_FLAG_MAX> _nnapi_flags;
+//   _nnapi_flags.set(NNAPI_FLAG_USE_FP16);
+//   unsigned long nnapi_flags = _nnapi_flags.to_ulong();
+enum NNAPIFlag : unsigned char {
```

Comment:
I am not sure if this file is the best place for the enums, these enums will be used by end users, and I don't want to put EP related setting into a common header such as common.h